### PR TITLE
Db/azure integration

### PR DESCRIPTION
**Pull request by davidbloss**
## Issues

[https://github.com/OpsLevel/team-platform/issues/414](https://github.com/OpsLevel/team-platform/issues/414) - Originally completed by @taimoorgit see [PR #430](https://github.com/OpsLevel/opslevel-go/pull/430)

## Changelog

- [ ] List your changes here
- [ ] Make a `changie` entry

### CODE REVIEW

File: input.go [hunk 1]
```diff
@@ -43,6 +43,17 @@ type AwsIntegrationInput struct {
 	AwsTagsOverrideOwnership *bool     `json:"awsTagsOverrideOwnership,omitempty" yaml:"awsTagsOverrideOwnership,omitempty" example:"false"`    // Allow tags imported from AWS to override ownership set in OpsLevel directly. (Optional.)
 }
 
+// AzureResourcesIntegrationInput specifies the input fields used to create and update an Azure resources integration.
+type AzureResourcesIntegrationInput struct {
+	Name                  *string   `json:"name,omitempty" yaml:"name,omitempty" example:"example_name"`                                        // The name of the integration. (Optional.)
+	OwnershipTagKeys      *[]string `json:"ownershipTagKeys,omitempty" yaml:"ownershipTagKeys,omitempty" example:"['tag_key1', 'tag_key2']"`    // An array of tag keys used to associate ownership from an integration. Max 5. (Optional.)
```
Comment (rocktavious) [inline input.go:49]: Just noticing this now but this is incorret.  The schema spec of this is `ownershipTagKeys: [String!]`. notice how on the inside of the array the non-null is on the string - not the array itself.  This means it should be

```suggestion
	OwnershipTagKeys      []*string `json:"ownershipTagKeys,omitempty" yaml:"ownershipTagKeys,omitempty" example:"['tag_key1', 'tag_key2']"`    // An array of tag keys used to associate ownership from an integration. Max 5. (Optional.)
```

File: integration.go [hunk 2]
```diff
@@ -19,6 +21,13 @@ type AWSIntegrationFragment struct {
 	OwnershipTagKeys     []string `graphql:"ownershipTagKeys"`
 }
 
+type AzureResourcesIntegrationFragment struct {
+	TenantId       string        `graphql:"tenantId"`
+	SubscriptionId string        `graphql:"subscriptionId"`
+	LastSyncedAt   *iso8601.Time `graphql:"lastSyncedAt"`
```
Comment (rocktavious) [inline integration.go:27]: i don't see this field in the public API

File: integration.go [hunk 3]
```diff
@@ -19,6 +21,13 @@ type AWSIntegrationFragment struct {
...
+	Aliases        []string      `graphql:"aliases"`
```
Comment (rocktavious) [inline integration.go:28]: additionally it seems like this is missing some of the fields from the input? 🤔

Comment (rocktavious) [reply integration.go:28]: Specifically
```
OwnershipTagKeys
TagsOverrideOwnership
```

File: integration_test.go [hunk 4]
```diff
@@ -43,10 +43,52 @@ func TestCreateAWSIntegration(t *testing.T) {
...
+		`mutation AzureResourcesIntegrationCreate($input:AzureResourcesIntegrationInput!){azureResourcesIntegrationCreate(input: $input){integration{id,name,type,createdAt,installedAt,... on AwsIntegration{iamRole,externalId,awsTagsOverrideOwnership,ownershipTagKeys},... on AzureResourcesIntegration{tenantId,subscriptionId,lastSyncedAt,aliases},... on NewRelicIntegration{baseUrl,accountKey}},errors{message,path}}}`,
+		`{"input": { "name": "new azure resources", "tenantId": "12345678-1234-1234-1234-123456789abc", "subscriptionId": "12345678-1234-1234-1234-123456789def", "clientId": "XXX_client_id_XXX", "clientSecret": "XXX_client_secret_XXX"}}`,
+		`{"data": {
+      "azureResourcesIntegrationCreate": {
+        "integration": {
+          {{ template "id1" }},
+          "name": "new azure resources",
+          "type": "azureResources",
+          "createdAt": "2024-07-04T16:25:29.574450Z",
+          "installedAt": "2024-07-04T16:25:28.541124Z",
```
Comment (rocktavious) [inline integration_test.go:58]: these are here but not in the `AzureResourcesIntegrationFragment` ?

File: integration_test.go [hunk 5]
```diff
@@ -43,10 +43,52 @@ func TestCreateAWSIntegration(t *testing.T) {
...
+          "tenantId": "12345678-1234-1234-1234-123456789abc",
+          "subscriptionId": "12345678-1234-1234-1234-123456789def",
+          "lastSyncedAt": null,
```
Comment (rocktavious) [inline integration_test.go:61]: this does not appear in the public API

File: integration.go (hunk 2 above)
Comment (davidbloss) [reply integration.go:27]: Same, this is out

File: integration.go (hunk 3 above)
Comment (davidbloss) [reply integration.go:28]: 👍 These are being added

File: integration_test.go (hunk 4 above)
Comment (davidbloss) [reply integration_test.go:58]: `createdAt` and `installedAt` are fields in the [Integration struct](https://github.com/OpsLevel/opslevel-go/pull/443/files#diff-b7b79f1ac09ee75185e8a967df69488f9fe9597d89aaa80d50296d9c028ee7d9R60-R61)

**Review by rocktavious — APPROVED** [review]

**Pull request merged**